### ISSUE TITLE
`rxCollCreate` returns different error codes based on presence of `RECURSIVE_OPR__KW` flag

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#7461]
- [ ] main
- [ ] 4-3-stable
---
## Refactor
Let's say we want to try and create a collection in a place where we don't have permission to. One may expect `rxCollCreate` to return the same permission error in all cases, but that's not what happens.

When `RECURSIVE_OPR__KW` is passed, you get `SYS_INVALID_INPUT_PARAM`.
When `RECURSIVE_OPR__KW` is NOT passed, you get `CAT_NO_ACCESS_PERMISSION`.

This was demonstrated using 4.3.1.

**Comment by alanking** [comment]
I don't think this is a refactor as a behavioral change is being proposed... Seems more like a bug to me as the expected behavior does not match the observed behavior. Thoughts?

**Comment by korydraughn** [comment]
Well, I don't think it's a bug. It's more of an enhancement IMO. I don't know that the original implementation was intended to have these operations result in identical error codes.

**Comment by korydraughn** [comment]
Definitely a change in behavior, which means this has to go into 4.4/5.0.

Consider whether it's worth adding a label for **behavioral change**.
